go/examples/main.go

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+package main
+
+import (
+	"fmt"
+	"github.com/aws/glide-for-redis/go/glide/glide"
+	"github.com/aws/glide-for-redis/go/glide/protobuf"
+)
+
+func main() {
+	fmt.Println("Starting go-glide client...")
+	client := glide.GlideRedisClient{}
```

Comment:
More consistent if we had a CreateClient() static method that called the constructor and returns:
```
client, error = glide.CreateClient(config);
```

Reply:
then we can make ConnectToRedis internal

Reply:
I have these changes in my connect branch and can change the go files to integrate with that branch in my upcoming connect PR. This PR should be focused more on the Rust code than the Go code since the Go code will change when I integrate